src/HTTPClient.ts

```diff
@@ -27,10 +27,41 @@ export class HTTPClient {
 	 * @public
 	 * @constructor
 	 * Creates an instance of a HTTPClient
-	 * @param {Middleware} middleware - The first middleware of the middleware chain
+	 * @param {Middleware| Middleware[]} middleware - The first middleware of the middleware chain or the array of middleware handlers
 	 */
-	public constructor(middleware: Middleware) {
-		this.middleware = middleware;
+	public constructor(middleware: Middleware | Middleware[]) {
+		this.setMiddleware(middleware);
+	}
+
+	/**
+	 * @private
+	 * Verifies if the middleware passed is an array or not before setting this.middleware property
+	 * @param {any} middleware - The middleware passed
+	 * @returns Nothing
+	 */
+
+	private setMiddleware(middleware: any): void {
+		if (Object.prototype.toString.call(middleware) === "[object Array]") {
```

Comment:
I think you can do `if Array.isArray(middleware)`

Reply:
I googled that `Array.isArray()` may not work in all environment and that is why used the condition above for array verification.

Reply:
Well I see now that Array.isArray is undefined in my browser.

Reply:
See my other suggestion which avoids testing it this way all together.
But overall we should either:
- Document a minimum required version of ECMAScript and encourage people to polyfill (like we're doing we Promise)
- Setup a pipeline to transpile down to different ES versions 

I don't think using "older ES" in the source code is going to make the maintenance any easier for us/get us good perf on modern engines.

Reply:
@baywet  It is a good point to set consider multiple ES versions during build/test phase. 
Regarding support for older ES,  I will look into that.